src/main/java/org/opensearch/commons/destination/message/LegacyBaseMessage.java

```diff
@@ -0,0 +1,127 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.commons.destination.message;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+/**
+ * This class holds the generic parameters required for a
+ * message.
+ */
+public abstract class LegacyBaseMessage implements Writeable {
+
+    protected LegacyDestinationType destinationType;
+    protected String destinationName;
+    protected String url;
+    private String content;
+
+    LegacyBaseMessage(final LegacyDestinationType destinationType, final String destinationName, final String content) {
+        if (destinationType == null) {
+            throw new IllegalArgumentException("Channel type must be defined");
+        }
+        if (!Strings.hasLength(destinationName)) {
+            throw new IllegalArgumentException("Channel name must be defined");
+        }
+        this.destinationType = destinationType;
+        this.destinationName = destinationName;
+        this.content = content;
+    }
+
+    LegacyBaseMessage(final LegacyDestinationType destinationType, final String destinationName, final String content, final String url) {
+        this(destinationType, destinationName, content);
+        if (url == null) {
+            throw new IllegalArgumentException("url is invalid or empty");
+        }
+        this.url = url;
+    }
+
+    LegacyBaseMessage(org.opensearch.common.io.stream.StreamInput streamInput) throws IOException {
+        this.destinationType = streamInput.readEnum(LegacyDestinationType.class);
+        this.destinationName = streamInput.readString();
+        this.url = streamInput.readOptionalString();
+        this.content = streamInput.readString();
+    }
+
+    public void setUrl(String url) {
+        this.url = url;
+    }
+
+    public LegacyDestinationType getChannelType() {
+        return destinationType;
+    }
+
+    public String getChannelName() {
+        return destinationName;
+    }
+
+    public String getMessageContent() {
+        return content;
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public URI getUri() {
+        return buildUri(getUrl().trim(), null, null, -1, null, null);
+    }
+
+    protected URI buildUri(String endpoint, String scheme, String host, int port, String path, Map<String, String> queryParams) {
```

Comment:
is this required to be in this class? this seems to be generic code which converts uri parts to uri

Reply:
This is just ported over from alerting notification subproject. Comments on previous implementation itself I am glossing over for now as the intention is to just move these classes here for ISM to use as it was previously without any modifications.